src/main/java/com/example/ludogorieSoft/village/controllers/AdministratorController.java

```diff
@@ -35,18 +38,62 @@ public ResponseEntity<AdministratorDTO> getAdministratorById(@PathVariable("id")
     public ResponseEntity<AdministratorDTO> createAdministrator(@Valid @RequestBody AdministratorRequest administratorRequest) {
         return ResponseEntity.ok(administratorService.createAdministrator(administratorRequest));
     }
+
     @PutMapping("/update/{id}")
     public ResponseEntity<AdministratorDTO> updateAdministrator(@PathVariable("id") Long id, @Valid @RequestBody AdministratorRequest administratorRequest) {
         return ResponseEntity.ok(administratorService.updateAdministrator(id, administratorRequest));
     }
+
     @DeleteMapping("/{id}")
     public ResponseEntity<String> deleteAdministratorById(@PathVariable("id") Long id) {
         administratorService.deleteAdministratorById(id);
         return new ResponseEntity<>("Administrator with id: " + id + " has been deleted successfully!!", HttpStatus.OK);
     }
 
     @GetMapping("village")
-    public List<VillageResponse> getAllVillages(){
-        return administratorService.getAllVillagesWithPopulation();
+    public List<VillageResponse> getAllVillages() {
+
+        return villageService.getAllVillagesWithAdmin();
+    }
+
+    @DeleteMapping("/village-delete/{villageId}")
+    public ResponseEntity<String> deleteVillageById(@PathVariable("villageId") Long villageId) {
+        villageService.deleteVillage(villageId);
+        return new ResponseEntity<>("Village with id: " + villageId + " has been deleted successfully!!", HttpStatus.OK);
+    }
+
+    @PostMapping("/approve/{villageId}")
+    public ResponseEntity<String> approveVillageResponse(@RequestParam("villageId") Long villageId,
+                                                      @RequestParam("answerDate") String answerDate) {
+        VillageDTO villageDTO = villageService.getVillageById(villageId);
+        villageService.updateVillageStatus(villageId, villageDTO);
+        adminVillageService.updateVillageStatusAndVillageResponsesStatus(villageId, answerDate);
+        return new ResponseEntity<>("Status of village with ID: " + villageId + " changed successfully!!!", HttpStatus.OK);
+    }
+
+    @GetMapping("/update/{villageId}")
+    public ResponseEntity<VillageDTO> findVillageById(@PathVariable(name = "villageId") Long id) {
+        VillageDTO village = villageService.getVillageById(id);
+
+        return new ResponseEntity<>(village, HttpStatus.OK);
+    }
+
+    @GetMapping("/update")
```

Comment:
Again URL and method names does not match.

Reply:
Yes, i fix it updateUnapprovedVillageResponseByVillageId().
Is that ok ?